go/arrow/csv/reader.go

```diff
@@ -721,6 +734,39 @@ func (r *Reader) parseDecimal256(field array.Builder, str string, prec, scale in
 	field.(*array.Decimal256Builder).Append(val)
 }
 
+func (r *Reader) parseList(field array.Builder, str string) {
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	if !(strings.HasPrefix(str, "{") || strings.HasSuffix(str, "}")) {
+		r.err = errors.New("invalid list format. should start with '{' and end with '}'")
+		field.AppendNull()
+		return
+	}
+	str = strings.TrimPrefix(str, "{")
+	str = strings.TrimSuffix(str, "}")
+	strs := strings.Split(str, ",")
```

Comment:
You probably need to use an actual `csv` reader here instead of just splitting on `","`. The reason is so that quotes are handled properly. Imagine a list of strings where one of the strings has a comma in them. Normal CSV handling would use quotes like `foo,bar,"something,baz"` to handle it. But if you just split on `","` it won't handle the quotes correctly.